bigbluebutton-html5/imports/ui/components/timer/panel/component.tsx

```diff
@@ -482,44 +485,71 @@ const TimerPanel: React.FC<TimerPanelProps> = ({
 
 const TimerPanelContaier: React.FC = () => {
   const [timerActivate] = useMutation(TIMER_ACTIVATE);
+  const { data: timerData } = useTimer();
+  const { data: currentUser } = useCurrentUser();
 
-  const {
-    data: timerData,
-  } = useTimer();
+  const amIModerator = !!currentUser?.isModerator;
 
   const activateTimer = useCallback(() => {
+  if ((timerData?.active) || !amIModerator) return;
+
     const TIMER_CONFIG = window.meetingClientSettings.public.timer;
-    const stopwatch = false;
-    const running = false;
-    const time = TIMER_CONFIG.time * MILLI_IN_MINUTE;
+    const meetingId = Auth.meetingID;
+    if (!meetingId) {
+      logger.warn({ logCode: 'timer_activate_no_meetingid' }, 'Timer activation skipped: no meetingId');
+      return;
+    }
+
+    timerActivate({
+      variables: {
+        stopwatch: false,
+        running: false,
+        time: TIMER_CONFIG.time * MILLI_IN_MINUTE,
+        meetingId,
+      },
+    }).catch((error) => {
+      if (!error.message?.includes('already active')) {
+        logger.error(
+          { logCode: 'timer_activate_failed', extraInfo: { errorMessage: error?.message } },
+          `Timer activation failed: ${error?.message}`
+        );
+      }
+    });
+  }, [timerActivate, timerData, amIModerator]);
+
+  useEffect(() => {
+    if ((timerData == null || timerData?.active === false) && amIModerator) {
```

Comment:
Ideally, `timerData` should never be `null`. If it is, it means the subscription hasn’t returned the timer data yet. In that case, we should wait for the data before trying to activate the timer.
```suggestion
    if (timerData?.active === false && amIModerator) {
```